web/profiles/custom/os2loop/modules/os2loop_lists/src/Helper/Helper.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\os2loop_lists\Helper;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+
+/**
+ * Os2Loop list helper.
+ *
+ * Helper class for creating content lists.
+ */
+class Helper {
+  /**
+   * The logged in user account.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The full user entity.
+   *
+   * @var \Drupal\user\UserStorageInterface
+   */
+  protected $userStorage;
+
+  /**
+   * Helper constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Entity type manager.
+   * @param \Drupal\Core\Session\AccountProxyInterface $currentUser
+   *   Logged in user account.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   *   Thrown if the entity type doesn't exist.
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   *   Thrown if the storage handler couldn't be loaded.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, AccountProxyInterface $currentUser) {
+    $this->userStorage = $entityTypeManager->getStorage('user');
+    $this->currentUser = $currentUser;
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Get content related by subject.
+   *
+   * @return array
+   *   An array of content that shares subject term with the logged in users
+   *   expertises terms.
+   */
+  public function getContentByUserExpertise(): array {
+    $user_expertises = $this->getCurrentUserExpertisesId();
+    if (empty($user_expertises)) {
+      return [];
+    }
+
+    return $this->entityTypeManager
+      ->getListBuilder('node')
+      ->getStorage()
+      ->loadByProperties([
+        'type' => 'os2loop_question',
+        'status' => 1,
+        'os2loop_shared_subject' => $user_expertises,
+      ]);
+  }
```

Comment:
You have almost the same code in both `getContentByUserExpertise` and `getContentByUserProfession`. Consider moving it into a helper function that can be called (twice).

Reply:
Added helper function